html-proofer: Remove error-sort flag to restore error verbosity

justwheel opened the pull request:
### Checklist

- [x] All new jobs, commands, executors, parameters have descriptions
- [x] Examples have been added for any significant new features
- [x] README has been updated, if necessary

### Motivation, issues

This commit removes the `error-sort` flag from the `htmlproofer` command
in order to restore logs from HTML Proofer in the Circle CI build
report.

The current behavior with this flag added is to suppress all error
messages, which is not helpful when you have a failing CI pipeline from
HTML Proofer. To debug requires a local checkout of the repo and to
manually run HTML Proofer to see the errors.

### Description

I tested this change locally and verified that removing this flag
restores the errors in log messages. This is an uncommon flag to use for
HTML Proofer, and I think it does not do exactly what it is expected to
do, so the best option appears to be removing it, thus simplifying what
needs to be maintained here.

justwheel commented:
Here is an example from my build pipeline, where `htmlproofer` reports 583 failures but didn't tell me what the root problem was (I needed to ignore internal cross-links from being checked, a feature I needed from #45):

https://app.circleci.com/pipelines/github/unicef/inventory/8/workflows/e76af7d9-1de4-471d-b8a2-31d3a3f50bf4/jobs/9

dsayling commented on src/commands/html-proofer.yml:
```diff
@@ -167,7 +163,6 @@ steps:
           --directory-index-file << parameters.directory-index-file >> \
           <<# parameters.disable-external >> --disable-external <</ parameters.disable-external >> \
           <<# parameters.empty-alt-ignore >> --empty-alt-ignore <</ parameters.empty-alt-ignore >> \
-          --error-sort << parameters.error-sort >> \
```
To maintain backward compatibility, you should add a boolean parameter, e.g.

```
  include-error-sort:
    description: "adds the --error-sort flag to the command"
    type: boolean
    default: true
 ```
 
 and then update this line,    `<<# parameters.include-error-sort >>--error-sort << parameters.error-sort >> <</ parameters.include-error-sort >> /`
 
This will not change existing usage for any user but would allow you to disable this feature.

gmemstr replied:
Better yet, we can leave the default error-sort parameter as blank and use that to evaluate the boolean.

```yaml
  error-sort:
    description:  "Defines the sort order for error output. Can be `:path`, `:desc`, or `:status` (default: `:path`)."
    type: string
    default: ""

#...

          <<# parameters.error-sort >>--error-sort << parameters.error-sort >> <</ parameters.error-sort >> \

```

justwheel replied:
Fixed in commit 2ce2d2917c0e51102d73cacf98a2b4113c724361.

justwheel committed:
:beetle: html-proofer: Adjust error-sort flag for better error verbosity

This commit changes the `error-sort` flag to use an empty parameter in
the `html-proofer` command. This causes HTML-Proofer to show specific
error messages in CircleCI build logs, where the current behavior is to
suppress error messages.

The current default behavior makes it difficult to debug errors from
HTML-Proofer in a failing CI pipeline. Debugging requires clone the
repository and running HTML-Proofer locally to see error messages.

This is an uncommon flag to use for HTML-Proofer, and I think it does
not do exactly what it is expected to do, so the best option appears to
be setting the parameter as blank.

Signed-off-by: Justin W. Flory (he/him) [UNICEF Innovation] <jflory@unicef.org>

justwheel commented:
Hi @dsayling and @gmemstr, apologies for the late follow-up here. But I implemented the suggested change in commit 2ce2d2917c0e51102d73cacf98a2b4113c724361, and I believe this is ready for final review. Could you please take a look? Thank you!

dsayling commented:
@jwflory - I'm no longer a maintainer here. @felicianotech, @KyleTryon

justwheel commented:
Hi @felicianotech and @KyleTryon, any chance you can help this move forward? Without this change, HTMLProofer prints no logs in CircleCI and makes it very hard to debug for people on our team who are not Hugo-savvy.
